src/compas/_json.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+import json
+from compas.utilities import DataEncoder
+from compas.utilities import DataDecoder
+
+
+__all__ = [
+    'json_dump',
+    'json_dumps',
+    'json_load',
+    'json_loads'
+]
+
+
+def json_dump(data, fp):
+    """Write a collection of COMPAS object data to a JSON file.
+
+    Parameters
+    ----------
+    data : list of :class:`compas.base.Base`
+        A list of COMPAS objects.
+    fp : file-like object or path
+        A writeable file-like object or the path to a file.
+
+    Returns
+    -------
+    None
+
+    Examples
+    --------
+    >>> import compas
+    >>> from compas.geometry import Point, Vector
+    >>> data1 = [Point(0, 0, 0), Vector(0, 0, 0)]
+    >>> compas.json_dump(data1, 'data.json')
+    >>> data2 = compas.json_load('data.json')
+    >>> data1 == data2
+    True
+    """
+    if hasattr(fp, 'write'):
+        return json.dump(data, fp, cls=DataEncoder)
+    with open(fp, 'w') as fp:
```

Comment:
If the plan is to fix up all of the file readers so that they take file-like objects and paths, I think these lines of code will be used quite a few times.  Maybe there should be a utility function which does this.  Some sort of context manager like 
```
class open_file(object):
    def __init__(self, fp, *args, **kwargs):
        if hasattr(fp, 'write'):
            self.f = fp
        else:
            self.f = open(fp, *args, **kwargs)

    def __enter__(self):
        return self.f

    def __close__(self):
        self.f.close()
        # but maybe you don't always want to close file-like objects...
```
And I suppose this could be generalized to handle reading as well.